### DISCUSSION TITLE
Per-Deploy Environment Variables

### DISCUSSION THREAD

**Discussion by gsheppard** [vercel/vercel#8756] [Help] [answered]
### Context
- Since I'm using Github Actions, I lose a lot of the perks of Vercel deployments, namely git branch information.
- I essentially have 4 environments I interact with: Production, Staging, PR Previews, and Development.
- Since Vercel only has a concept of 3 of those, in order to manually recreate PR Previews I've been reliant on `VERCEL_ENV==='preview' && GIT_BRANCH` to determine that it's a PR Preview and not Staging.
- This works at build time since I'm determining a custom `GIT_BRANCH` env var via Github Actions, but I lose it in runtime (trying to access in NextJS middleware)

### Question
- When using Github Actions and manually deploying via the Vercel CLI, what's the best way to apply per-deployment environment variables (such as a custom `GIT_BRANCH`) in such a way that they are available at runtime?

**Comment by amyegan** [comment, marked as answer]
Hi @gsheppard! When using Vercel CLI, you can use the [`--env`](https://vercel.com/docs/cli/deploy#env) option to provide runtime environment variables.

**Reply by dakdevs** [reply]
Is there anyway to achieve this from a env file? `--env-file`? Then at least we can use the same `.env.preview.local` that the `vercel build` command uses.
